src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -262,20 +262,20 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         if (_connection)
         {
             _connectionOutputEventRevoker.revoke();
+
+            // Fire off a connection state changed notification, to let our hosting
+            // app know that we're in a different state now.
+            if (newConnection.State() != _connection.State())

```

Comment:
wait, this has to go into the block **after** `_connection` changes (I think).

This is because somebody will call right back into TermControl to ask the current state (!)

At this point it will reflect the old connection.